processor/src/test/java/org/mapstruct/ap/test/bugs/_938/Issue938Mapper.java

```diff
@@ -0,0 +1,32 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.bugs._938;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+/**
+ * @author yokotaso <yokotaso.t@gmail.com>
+ */
+@Mapper(componentModel = "spring", uses = Issue938Service.class)
+public interface Issue938Mapper {
+
+    @Mapping(target = "value", expression = "java(issue938Service.convert( source ))")
```

Comment:
I'm not in favour of directly referring to an generated name in an expression. For one thing: the internal workings of MapStruct (and naming of fields) are up for change when we add more functionality. However, we could create some syntax to do this more logically by referring to the used mapper type. E.g. " java( ${Issue938Service.class}.convert( source ) )" 

More general: I'm doubting whether this kind of functionality is still required now we have the `@Context` annotation. You can bring a mapper to a handwritten method by means of context. 

I will also put this up as a comment in the issue.

@gunnarmorling : WYDT?

Reply:
So personally I'm not too concerned about referring to the fields from within expressions. It's not that we'd arbitrarily change it, but more importantly, the generated code wouldn't compile if there was a mismatch, so one would find out about it very quickly.

The bigger problem I see is how to detect the usage within an expression reliably. The proposed way of simply checking `contains()` may lead to superfluous fields being added, when e.g. having the name of a field within a string literal within the expression. OTOH it's not a realy problem I suppose. Ultimately, it leads back to the idea to having a proper expression language within `expression`, which we'd then fully parse and thus understand what are actual field references. But that's not something right around the corner :)

The context solution seems workable, too. It might actually be the better way to do it.